### PR TITLE
Convert `Shw::Outliner` component to GTS

### PR DESCRIPTION
**Pull request by zamoore**
### :pushpin: Summary

If merged, this PR converts the `Shw::Outliner` component to GTS.

### :link: External links

Jira ticket: [HDS-3836](https://hashicorp.atlassian.net/browse/HDS-3836)

***

### 👀 Component checklist

- [ ] Percy was checked for any visual regression
- [ ] A changelog entry was added via [Changesets](https://github.com/changesets/changesets) if needed (see [templates here](https://hashicorp.atlassian.net/wiki/spaces/HDS/pages/3243114706/Changelog+authoring+best+practices#Templates))

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

[HDS-3836]: https://hashicorp.atlassian.net/browse/HDS-3836?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by didoo** [comment]
@zamoore you also have to update the template registry and add an `index` to the import:
https://github.com/hashicorp/design-system/pull/2412/files#diff-cc7c68c95e27cf8cd585b72e73e6dc3038eee3982b44d44b5b5a33bcd467da03

**Comment by zamoore** [comment]
@didoo - thanks, let me know if that change looks right

**Review by didoo — APPROVED** [review]
LGTM 👍

File: showcase/app/components/shw/outliner/index.gts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+import Component from '@glimmer/component';
+
+export interface ShwOutlinerSignature {
+  Blocks: {
```
Comment (aklkv) [inline showcase/app/components/shw/outliner/index.gts:8]: just FYI: not having even empty `Args` block here might break types in the context of an other `.gts` component, but it's fine for now

**Pull request merged**